microservices/realtime/src/Services/RegistrationChannelResolver.php

```diff
@@ -32,18 +32,53 @@ public function __construct(
         $this->ddiProviderRepository = $ddiProviderRepository;
     }
 
+    public function getDefaultCriteria(array $tokenPayload): array
+    {
+        $this->testDbConnnection();
+        $username = $tokenPayload['username'] ?? null;
+        if (!$username) {
+            throw new \Exception(
```

Comment:
It is not a better idea to use DomainException instead?

Reply:
yes it is